### ISSUE TITLE
Need a solution for non-distinguishable links

### ISSUE THREAD

**Issue by austinstarin** [apostrophecms/apostrophe-link-widgets#7]
Should this be an aria-labelledby option or hidden text option?

cc/ @arti5m @bgantick @mtthwmnc @abea

**Comment by abea** [comment]
First I'm hearing [that term](https://oit.boisestate.edu/blog/2017/07/19/spotlight-web-accessibility-non-distinguishable-links/). From a bit of searching it seems that `aria-labelledby` might be appropriate and specifically meant to help in this kind of situation.

**Comment by bgantick** [comment]
I think this would be putting a band-aid on a problem that shouldn't exist. By providing an option to "fix" poorly constructed links, we're reinforcing bad habits.

**Comment by austinstarin** [comment]
@bgantick can you unpack that? are you saying this should be considered from a design perspective?

**Comment by bgantick** [comment]
I'm saying links should always be descriptive. If we were to add an option, I would agree with @abea that `aria-labelledby` would be appropriate: https://www.w3.org/WAI/GL/wiki/Using_aria-label_for_link_purpose. I want to reiterate that adding this option is a crutch and is opening the door for folks to misuse/rely on it.

**Comment by austinstarin** [comment]
That makes sense. I run into issues where there are "learn more" buttons or "read more" buttons that need special care.

**Comment by bgantick** [comment]
Also: I'm inclined to say that if folks aren't bothering to make links descriptive, they might not take the time to fill out an additional field.

**Comment by austinstarin** [comment]
Good point... Although that's like saying "let's not do accessibility, it's too hard for people".

**Comment by bgantick** [comment]
Looking deeper at the official spec: https://www.w3.org/TR/wai-aria-1.1/#aria-labelledby - this might be hard to implement on the widget level. `aria-labelledby` actually looks for an element to label the current element by (like a heading). Example:

```
<h2 id="headline">Storms hit east coast</h2>

<p>Torrential rain and gale force winds have struck the east coast, causing flooding in many coastal towns.
<a id="p123" href="news.html" aria-labelledby="p123 headline">Read more...</a></p>
```
Maybe `aria-label` would be more appropriate?

**Comment by abea** [comment]
I'm starting to think hidden text might be better. `aria-label` is described as for when there's no text because the idea is that if text is present it should be descriptive. Even "Read more" might be fine in situations where there's teaser text and/or a header before it.

So a bullet-proof (but not really) solution might be like:
`<a href="..." ><span class="visually-hidden">This is what it's about</span><span aria-hidden="true">Read more</span></a>`

That hurt to write. So maybe we should have something like that? But it is a fair point that accessibility shouldn't be left until the last stage in website building. Design and content strategy should probably take care of this problem before it becomes one.

**Comment by abea** [comment]
Nevermind. Brian's right about aria-label. In my DMs dropping w3 wiki articles.

**Comment by bgantick** [comment]
From the w3c wiki (linked previously: https://www.w3.org/WAI/GL/wiki/Using_aria-label_for_link_purpose):

>Describing the purpose of a link in HTML using the aria-label element. In some situations, designers may choose to lessen the visual appearance of links on a page by using shorter, repeated link text such as "read more". These situations provide a good use case for aria-label in that the simpler, non-descriptive "read more" text on the page can be replaced with a more descriptive label of the link. The words 'read more' are repeated in the aria-label (which replaces the original anchor text of "[Read more...]") to allow consistent communication between users.

¯\_(ツ)_/¯

**Comment by austinstarin** [comment]
See also here for "Read More" links https://a11yproject.com/patterns#anchors-links

@arti5m Didn't you have a strong opinion about this?
